### PR TITLE
update mime type and extension

### PR DESCRIPTION
**Pull request by yosef32**
When converting a DOC document to DOCX then you need to change the extension and the mime type. And that makes it possible to do that

### update 11/10/2020
**run on my environment http://yosef.northeurope.cloudapp.azure.com
and confirm its work by lihi from barak team**

### CODE REVIEW

File: upload/update.go [hunk 1]
```diff
@@ -234,3 +257,21 @@ func (r *Router) deleteUpdateOnError(c *gin.Context, err error, upload *fpb.GetU
 	httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
 	loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
 }
+
+// changeExtensionByMimeType return the same file name and change the extention by the mime type
```
Comment (Shahar-Y) [inline upload/update.go:261]:
```suggestion
// changeExtensionByMimeType returns the same file name and changes the extension by the mime type
```

File: upload/update.go [hunk 2]
```diff
@@ -234,3 +257,21 @@ func (r *Router) deleteUpdateOnError(c *gin.Context, err error, upload *fpb.GetU
...
+// It is only for docs
```
Comment (Shahar-Y) [inline upload/update.go:262]:
```suggestion
// This function is only used for the docs
```

**Review by Shahar-Y — CHANGES_REQUESTED** [review]
Run it in the k8s and add a link

File: upload/update.go [hunk 3]
```diff
@@ -234,3 +257,21 @@ func (r *Router) deleteUpdateOnError(c *gin.Context, err error, upload *fpb.GetU
...
+		}
+		case MimeTypePPTX: {
+			splitName[len(splitName) - 1] = "pptx"
+		}
+		case MimeTypeXLSX: {
+			splitName[len(splitName) - 1] = "xlsx"
+		}
+	}
+	return strings.Join(splitName, ".")
+}
```
Comment (Shahar-Y) [inline upload/update.go:277]: This overrides the last part of the array. 
Check if there is an extension name, if yes -> override it, if not-> add the extension

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

File: upload/update.go (hunk 3 above)
Comment (yosef32) [reply upload/update.go]: done

File: upload/update.go [hunk 4]
```diff
@@ -161,10 +174,20 @@ func (r *Router) UpdateComplete(c *gin.Context) {
 		return
 	}
 
+	mimeType := c.GetHeader(MimeTypeHeader)
+	fileName := oldFile.Name
+	if mimeType == "" {
+		mimeType = resp.GetContentType()
```
Comment (Shahar-Y) [inline upload/update.go:180]: How can there be two mime type sources? (c.GetHeader(MimeTypeHeader), resp.GetContentType())

**Review by Shahar-Y — CHANGES_REQUESTED** [review]
Moving to changes required until testing with Docs are approved

File: upload/update.go (hunk 3 above)
Comment (Shahar-Y) [reply upload/update.go]: Then it needs to be tested before review

File: upload/update.go (hunk 4 above)
Comment (yosef32) [reply upload/update.go:180]: When you update a regular file it returns mime type from S3, and when you change only the mime type then it is sent in the headers

**Pull request merged**